e2e/scaffolding/autoNonce.ts

```diff
@@ -11,24 +11,122 @@ import { getUnifiedAddress } from './ethereum';
 
 export type AutoNonce = number | 'auto' | 'current';
 
-const nonceCache = new Map<string, number>();
```

Comment:
Want to learn more about Atomics? Check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics

Reply:
Is it because there is not get and set singular operation in the map so that it could switch the context after getting in A and get the same value in B and end up with the same nonce?

Reply:
Hmm, interesting. I wouldn't have thought this would have a benefit in Node unless using WebWorkers, due to the single-threaded nature... but due to some async operations themselves yielding flow control due to awaiting secondary async operations, I guess it can still be useful/necessary.